tests/parametric/test_span_links.py

```diff
@@ -73,20 +85,27 @@ def test_span_link_from_distributed_datadog_headers(self, test_agent, test_libra
         representation in span links.
         """
         with test_library:
-            with test_library.start_span(
-                "root",
-                links=[Link(parent_id=0, attributes={"foo": "bar"})],
-                http_headers=[
-                    ["x-datadog-trace-id", "1234567890"],
-                    ["x-datadog-parent-id", "9876543210"],
-                    ["x-datadog-sampling-priority", "2"],
-                    ["x-datadog-origin", "synthetics"],
-                    ["x-datadog-tags", "_dd.p.dm=-4,_dd.p.tid=0000000000000010"],
-                ],
-            ):
+            with test_library.start_span("first") as first:
                 pass
 
-        span = get_span(test_agent)
+                with test_library.start_span(
+                    "remote_span",
+                    http_headers=[
+                        ["x-datadog-trace-id", "1234567890"],
+                        ["x-datadog-parent-id", "9876543210"],
+                        ["x-datadog-sampling-priority", "2"],
+                        ["x-datadog-origin", "synthetics"],
+                        ["x-datadog-tags", "_dd.p.dm=-4,_dd.p.tid=0000000000000010"],
+                    ],
+                ):
+                    pass
+
+                with test_library.start_span("second") as second:
+                    pass
+
+        traces = test_agent.wait_for_num_traces(1)
+
+        span = traces[0][2]
```

Comment:
I don' understand this test. The original test ask to create a span links from HTTP headers (using the original way, it has changed to use the headers field of the link object).

Here, you create a first span.
Then a second span from http headers,
Finally, a last span (called second). This one is supposed to have a span link? And used the second created span headers? Why is that?

Reply:
The naming is confusing. I should have named the spans: "parent", "linked_span", "child". The goal here is to create an integration test for how span links would be set in a normal application. This is the flow I am trying to recreate: 
 - Span 1 is open (root/parent) -> distributed tracing headers containing span 2 are received (first/linked_span) -> Span 3 is created (second/child). Span 3 is a child of Span 1. -> Span 1 and Span 3 are ended

In this flow Span 3 is a child of Span 1 and is LINKED to Span 2. With this flow span links are never set by the test case. It is set in the application. I can refactor this test to make these relationships more clear. I am noticing a bug that is making this test hard to follow. Does this flow make sense? Is this what we should be doing?